File: python/01-Fundamentals/01-fundamentals.py
#1) Setting and Swapping
def swapValues():
    myNumber = 42
    myName = 'Cameron'
    print(myNumber)
    print(myName)
    temp = myNumber
    myNumber = myName
    myName = temp
    print(myNumber)
    print(myName)
#swapValues()

#2) Print and Count
def printMultiplesOf5():
    sum = 0
    for i in range(5, 4096, 5):
        print(i)
        sum += 1
    print(sum)
#printMultiplesOf5()

#3) Print -52 to 1066
def printInt():
    for i in range(-52, 1067):
        print(i)
#printInt()

#4) Multiples of Six
def miultiplesOfSix():
    i = 6
    while i <= 60000:
        print(i)
        i += 6
#miultiplesOfSix()

#5) Don’t Worry, Be Happy
def beCheerful():
    for i in range(99):
        print("Number {}: Good morning!".format(i))
#beCheerful()

#6) Counting, the Dojo Way
def dojoCounting():
    for i in range(101):
        if i % 10 == 0:
            print("Coding Dojo")
        elif i % 5 == 0:
            print("Coding")
        else:
            print(i)
#dojoCounting()

#7) Multiples of Three – but Not All
def multiplesOfThree():
    for i in range(-300, 0, +3):
        if i == -6 or i == -3:
            continue
        else:
            print(i)
#multiplesOfThree()

#8) What Do You Know?
def printIncoming(input):
    print(input)
variable = "parameter"
#printIncoming(variable)

#9) Printing Integers with While
def printWithWhile():
    i = 2000
    while i <= 5280:
        print(i)
        i += 1
#printWithWhile()

#10) Whoa, That Sucker’s Huge...
def whoaBigNum():
    sum = 0
    for i in range(1, 300000, +2):
        sum += i
    return sum*2
#print(whoaBigNum())

#11) You Say It’s Your Birthday
def cakeDay(num1, num2):
    if num1 == 12 or num1 == 21:
        if num2 == 12 or num2 == 21:
            print("How did you know!?")
        else:
            print("Just another day....")
    else:
            print("Just another day....")
day = 12
month = 21
#cakeDay(day, month)

#12) Countdown by Fours
def countDown():
    i = 2016
    while i > 0:
        print(i)
        i -= 4
#countDown()

#13) Leap Year
def leapYear(input):
    if input % 4 == 0:
        if input % 100 == 0:
            if input % 400 == 0:
                print("A Special Kind of Leap Year!")
            else:
                print("Not a Leap Year.")
        else:
            print("Leap Year!")
    else:
        print("Not a Leap Year.")
year = 2000
#leapYear(year)

#14) Flexible Countdown
def flexibleCountdown(min, max, mult):
    for i in range(max, min, -mult):
        print(i)
flexibleCountdown(2,9,3)

#15) The Final Countdown
def finalCountdown(mult, min, max, exclude):
    while min <= max:
        if min % mult ==0:
            if min == exclude:
                min +=1
                continue
            print(min)
        min += 1
#finalCountdown(3,5,17,9)

File: javaScript/01-Fundamentals/01-fundamentals.js
'use strict';
// 1
var myNumber = 42;
var myName = 'Cameron';
var temp;
// 2
function swapMyNameNum(){
    temp = myName;
    myName = myNumber;
    myNumber = temp;
    console.log('myName: '+myName);
    console.log('myNumber:'+myNumber);
}
//swapMyNameNum();

// 3
function printInt(){
    for(var i = -52; i <= 1066; i++){
        console.log(i);
    }
}
//printInt();

// 4
function printAndCount(){
    var count = 0;
    for(var i = 512; i <=4096; i++){
        if(i % 5 == 0){
            console.log(i);
            count += 1;
        }
    }
    return count;
}
//console.log(printAndCount());

// 5
function multiplesOf6(){
    var i = 0;
    while(i <= 60000){
        console.log(i);
        i += 6;
    }
}
//multiplesOf6();

// 6
function beCheerful(){
    var greeting = 'good morning!'
    for(var i = 1; i <= 98; i++){
        console.log(i+":"+greeting);
    }
}
//beCheerful();

// 7
function ninjaCounting(){
    for(var i = 1; i <= 100; i++){
        if(i % 10 == 0){
            console.log("Coding Dojo");
        }
        else if(i % 5 == 0){
            console.log("Coding");
        }
        else{
            console.log(i);
        }
    }
}
//ninjaCounting();

// 8
function multiplesOf3(){
    for(var i = -300; i <= 0; i+=3){
        if(i == -6 || i == -3){
            continue;
        }
        console.log(i);
    }
}
//multiplesOf3();

// 9
function printParameter(input){
    console.log(input);
}
//printParameter("Hello World");

// 10
function printWhile(){
    var num = 2000
    while (num <= 5280) {
        console.log(num);
        num++;
    }
}
//printWhile();

// 11
function largeNum(){
    var i = 1;
    var sum = 0;
    while(i <= 299999){
        sum += i;
        i += 2;
    }
    console.log(sum*2);
}
//largeNum();

// 12
function specialDay(num1, num2){
    if(num1 == 12 || num1 == 21 && num2 == 12 || num2 == 21){
        console.log("How did you know?")
    }else{
        console.log("Just another day...");
    }
}
specialDay(12, 10);

// 13
function fourDown(){
    var i = 2016;
    while(i > 0){
        console.log(i);
        i -= 4;
    }
}
//fourDown();

// 14
function flexibleCountdown(max, min, mult){
    for(var i = max; i > min; i -= mult){
        console.log(i);
    }
}
//flexibleCountdown(9,2,3);

// 15
function leapyear(input){
    if(input % 4 ==0){
        if(input % 100 == 0){
            if(input % 400 == 0){
                console.log("Special Leap Year.")
            }else{
                console.log("Not a Leap Year.")
            }
        }else{
            console.log("Leap Year.")
        }
    }else{
        console.log("Not a Leap Year.")
    }
}
//leapyear(2020);

// 16
function finalCountdown(mult, from, to, except){
    var i = from;
    while(i <= to){
        if(i % mult == 0){
            if(i == except){
                i ++;
                continue
            }
            console.log(i);
        }
        i ++;
    }
}
//finalCountdown(3,5,17,9);
